massa-grpc/src/stream/new_operations.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023 MASSA LABS <info@massa.net>
+
+use crate::error::GrpcError;
+use crate::server::MassaGrpc;
+use futures_util::StreamExt;
+use massa_models::operation::OperationType;
+use massa_proto::massa::api::v1 as grpc;
+use std::pin::Pin;
+use tokio::select;
+use tonic::codegen::futures_core;
+use tonic::{Request, Streaming};
+use tracing::log::error;
+
+/// Type declaration for StreamTransactionsThroughputStream
+pub type NewOperationsStream = Pin<
+    Box<
+        dyn futures_core::Stream<Item = Result<grpc::NewOperationsStreamResponse, tonic::Status>>
+            + Send
+            + 'static,
+    >,
+>;
+
+/// Creates a new stream of new produced and received operations
+pub(crate) async fn new_operations(
+    grpc: &MassaGrpc,
+    request: Request<Streaming<grpc::NewOperationsStreamRequest>>,
+) -> Result<NewOperationsStream, GrpcError> {
+    // Create a channel to handle communication with the client
+    let (tx, rx) = tokio::sync::mpsc::channel(grpc.grpc_config.max_channel_size);
+    // Get the inner stream from the request
+    let mut in_stream = request.into_inner();
+    // Subscribe to the new operations channel
+    let mut subscriber = grpc.pool_channels.operation_sender.subscribe();
+
+    // Spawn a new task for sending new operations
+    tokio::spawn(async move {
+        if let Some(Ok(request)) = in_stream.next().await {
+            let mut request_id = request.id;
+            let mut filter = request.query.and_then(|q| q.filter);
+
+            // Spawn a new task for sending new blocks
+            loop {
+                select! {
+                    // Receive a new operation from the subscriber
+                     event = subscriber.recv() => {
+                        match event {
+                            Ok(operation) => {
+                                match operation.clone().content.op {
+                                    OperationType::Transaction{recipient_address,amount} => {
+                                        if !should_send(&filter, grpc::OperationTypeEnum::Transaction) {
+                                            continue
+                                        }
+                                        grpc::OperationType{
```

Comment:
Why you don't use the `From` declared here : https://github.com/massalabs/massa/pull/3443/files#diff-8d388876bf9b95596cca493e9b26d724994f068637fdf817b4de58494791e6aaR105 ?

Reply:
Hm thx , forget to remove grpc::OperationType in the match. 

From is now used at line 114.

Reply:
Maybe you didn't push the commit or I don't see it ?

Reply:
i forget to remove it, i'm on it

Reply:
@AurelienFT 
Done here : https://github.com/massalabs/massa/pull/3443/commits/d435bc2548fa99c0212e7b0311a9c203eb0491be